web/packages/teleterm/src/services/tshd/createClient.ts

```diff
@@ -903,105 +955,108 @@ export function createTshdClient(
-              const res = response.toObject();
-              resolve(res.userPreferences);
+              resolve(response!.userPreferences!);
             }
           });
         });
       });
     },
     updateUserPreferences(
-      params: api.UpdateUserPreferencesRequest.AsObject,
+      params: api.UpdateUserPreferencesRequest,
       abortSignal?: types.TshAbortSignal
-    ): Promise<api.UserPreferences.AsObject> {
-      const userPreferences = new UserPreferences();
-      if (params.userPreferences.clusterPreferences) {
-        userPreferences.setClusterPreferences(
-          new ClusterUserPreferences().setPinnedResources(
-            new PinnedResourcesUserPreferences().setResourceIdsList(
-              params.userPreferences.clusterPreferences.pinnedResources
-                .resourceIdsList
-            )
-          )
-        );
-      }
+    ): Promise<api.UserPreferences> {
+      const userPreferences = UserPreferences.create();
+      if (params.userPreferences) {
+        if (params.userPreferences.clusterPreferences) {
+          userPreferences.clusterPreferences = ClusterUserPreferences.create({
+            pinnedResources: PinnedResourcesUserPreferences.create({
+              resourceIds:
+                params.userPreferences.clusterPreferences.pinnedResources
+                  ?.resourceIds,
+            }),
+          });
+        }
 
-      if (params.userPreferences.unifiedResourcePreferences) {
-        userPreferences.setUnifiedResourcePreferences(
-          new UnifiedResourcePreferences()
-            .setDefaultTab(
-              params.userPreferences.unifiedResourcePreferences.defaultTab
-            )
-            .setViewMode(
-              params.userPreferences.unifiedResourcePreferences.viewMode
-            )
-            .setLabelsViewMode(
-              params.userPreferences.unifiedResourcePreferences.labelsViewMode
-            )
-        );
+        if (params.userPreferences.unifiedResourcePreferences) {
+          userPreferences.unifiedResourcePreferences =
+            UnifiedResourcePreferences.create({
+              defaultTab:
+                params.userPreferences.unifiedResourcePreferences.defaultTab,
+              viewMode:
+                params.userPreferences.unifiedResourcePreferences.viewMode,
+              labelsViewMode:
+                params.userPreferences.unifiedResourcePreferences
+                  .labelsViewMode,
+            });
+        }
       }
 
       return withAbort(abortSignal, callRef => {
-        const req = new api.UpdateUserPreferencesRequest()
-          .setClusterUri(params.clusterUri)
-          .setUserPreferences(userPreferences);
+        const req = api.UpdateUserPreferencesRequest.create({
+          clusterUri: params.clusterUri,
+          userPreferences: {
+            unifiedResourcePreferences:
+              userPreferences.unifiedResourcePreferences,
+            clusterPreferences: userPreferences.clusterPreferences,
```

Comment:
I think we can pass `userPreferences` directly.

Reply:
Yep!